### PR TITLE
Feat/dgds 302 specify onclick actions mapboxlayers

### PR DESCRIPTION
**Pull request by CindyvdVries**

### CODE REVIEW

File: components/v-mapbox-components/v-mapbox-selected-point-layer.vue [hunk 1]
```diff
@@ -43,13 +43,15 @@ export default {
     geometry(newValue) {
       const selectedLayer = this.map.getSource(this.selectedLayer.id)
       selectedLayer.setData(newValue)
+      this.map.moveLayer(this.selectedLayer.id)
```
Comment (SiggyF) [inline components/v-mapbox-components/v-mapbox-selected-point-layer.vue]: Move layer to top?

File: components/v-mapbox-components/v-mapbox-vector-layer.vue [hunk 2]
```diff
@@ -20,6 +24,7 @@ export default {
   data() {
     return {
       map: null,
+      layerToZoomTo: '',
```
Comment (SiggyF) [inline components/v-mapbox-components/v-mapbox-vector-layer.vue]: null/''?

File: components/v-mapbox-components/v-mapbox-vector-layer.vue [hunk 3]
```diff
@@ -93,6 +78,82 @@ export default {
...
+    showGraph(layer, event) {
+      this.adjustMap(event, 500)
+
+      const features = this.map.queryRenderedFeatures(event.point)
+      this.$emit('select-locations', {
+        features,
+        geometry: features[0].geometry,
+      })
+    },
+    adjustMap(event, duration) {
```
Comment (SiggyF) [inline components/v-mapbox-components/v-mapbox-vector-layer.vue]: adjust?

File: components/v-mapbox-components/v-mapbox-vector-layer.vue [hunk 4]
```diff
@@ -93,6 +78,82 @@ export default {
...
+      // features of the layertozoomto
+      const zoomLevel = _.get(layer, 'onClick.zoomTo')
+      const duration = 500
+
+      this.map.flyTo({
+        center: event.lngLat,
+        zoom: zoomLevel,
+      })
+
+      setTimeout(() => {
```
Comment (SiggyF) [inline components/v-mapbox-components/v-mapbox-vector-layer.vue]: why  timeout?

Comment (SiggyF) [reply components/v-mapbox-components/v-mapbox-vector-layer.vue]: Or you could to something like this, perhaps in v-mapbox
```
function flyTo(lngLat, zoom) {
 let flyTime = 300
 let promise = new Promise(function(resolve, reject) {
  this.map.flyTo()
  setTimeout(function() {
    resolve()
  }, flyTime)
 })
 return promise
})
```

en dan 
`await this.flyTo()`

File: components/v-mapbox-components/v-mapbox-vector-layer.vue [hunk 5]
```diff
@@ -93,6 +78,82 @@ export default {
...
+        const index = distances.indexOf(Math.min.apply(null, distances))
+        const nextFeat = feats[index]
+        this.$emit('select-locations', {
+          features: [nextFeat],
+          geometry: nextFeat.geometry,
+        })
+      }, duration * 9)
+    },
+
+    getBbox(lngLat) {
```
Comment (SiggyF) [inline components/v-mapbox-components/v-mapbox-vector-layer.vue]: getBBox?

File: components/v-mapbox-components/v-mapbox-selected-point-layer.vue (hunk 1 above)
Comment (CindyvdVries) [reply components/v-mapbox-components/v-mapbox-selected-point-layer.vue]: Yes, this is the layer indicating the selection. Which was now each time underneath the mapboxlayer, which made it hard to see which point was selected.

File: components/v-mapbox-components/v-mapbox-vector-layer.vue (hunk 2 above)
Comment (CindyvdVries) [reply components/v-mapbox-components/v-mapbox-vector-layer.vue]: This needs to be removed, not used anymore

File: components/v-mapbox-components/v-mapbox-vector-layer.vue (hunk 3 above)
Comment (CindyvdVries) [reply components/v-mapbox-components/v-mapbox-vector-layer.vue]: changed to mapPanTo

File: components/v-mapbox-components/v-mapbox-vector-layer.vue (hunk 4 above)
Comment (CindyvdVries) [reply components/v-mapbox-components/v-mapbox-vector-layer.vue]: I think it is a good idea to put it in v-mapbox

File: components/v-mapbox-components/v-mapbox-vector-layer.vue (hunk 5 above)
Comment (CindyvdVries) [reply components/v-mapbox-components/v-mapbox-vector-layer.vue]: check

**Pull request merged**